connectors/xai_connector.py

```
@@ -48,159 +48,157 @@ def __init__(self, api_key: Optional[str] = None):
                 result = response.json()
-                
+
                 # Format response with Material Design 3 structure
                 return self._format_md3_response(result)
-                
+
             except httpx.HTTPStatusError as e:
-                logger.error(f"xAI API error: {e.response.status_code} - {e.response.text}")
+                logger.error(
+                    f"xAI API error: {e.response.status_code} - {e.response.text}"
+                )
```

Ruff (0.11.9)

Use `logging.exception` instead of `logging.error`

Replace with `exception`

(TRY400)